osu.Game/Graphics/Sprites/OsuSpriteText.cs

```diff
@@ -20,6 +21,31 @@ public OsuSpriteText()
             TextSize = FONT_SIZE;
         }
 
+        public void Truncate(float maxWidth, bool ellipsis = false)
+        {
+            if (ellipsis)
+            {
+                Texture period = this.GetTextureForCharacter('.');
+                // Scale the width to the current TextSize
+                float scaleFactor = (float)period.Width / period.Height;
+                maxWidth -= 3 * (period.Width - (period.Height - this.TextSize) * scaleFactor);
+
+            }
+
+            float totalWidth = 0;
+            for (int a = 0; a < this.Text.Length; a++)
+            {
+                Texture t = this.GetTextureForCharacter(this.Text[a]);
```

Comment:
This seems expensive.

Reply:
It does look expensive, but because there is no way (that I was able to find) to get the width of the container itself this was the only solution I could find. I admittedly didn't run any performance tests, but I can tell you that it appeared to have no visual effect on my (above average) computer.

Reply:
Ran diagnostics:
The average of 10 tests (1,000,000 calls) was 612.8 milliseconds, which comes out to 0.0006128 milliseconds per call, or 0.009192 milliseconds per user (assuming 15 character username).
This assumes equal time lookup for any character.
```c#
public void perfTest()
{
    for (int a = 0; a < 10; a++)
    {
        Stopwatch stopwatch = Stopwatch.StartNew();
        for (int b = 0; b < 1000000; b++)
        {
            Texture period = GetTextureForCharacter('.');
        }
        stopwatch.Stop();
        System.Console.WriteLine("TIME: " + stopwatch.ElapsedMilliseconds);
    }
}
```